lib/api/src/modules/refs.ts

```diff
@@ -0,0 +1,205 @@
+      knownAs: id,
+      // this is used to know which iframe to emit the message to
+      refId: ref.id,
+    };
+
+    Object.assign(target, item, addition);
+
+    if (!isRoot(item)) {
+      const mappedParentId = `${ref.id}_${item.parent}`;
+
+      Object.assign(target, {
+        parent: mappedParentId,
+      });
+    }
+
+    if (item.children) {
+      Object.assign(target, {
+        children: item.children.map(c => `${ref.id}_${c}`),
+      });
+    }
+    output[mappedId] = target;
+  });
+
+  return output;
+};
+
+const map = (input: StoriesRaw, ref: InceptionRef, options: { mapper?: Mapper }): StoriesRaw => {
+  const output: StoriesRaw = {};
+  // map the incoming stories to a prefixed, non-conflicting version
+  Object.entries(input).forEach(([unmappedStoryId, unmappedStoryInput]) => {
+    const mapped = options.mapper ? options.mapper(ref, unmappedStoryInput) : unmappedStoryInput;
+
+    if (mapped) {
+      output[unmappedStoryId] = mapped;
+    }
+  });
+  return output;
+};
+
+const initRefsApi = ({ store, provider }: Module) => {
+  const api: SubAPI = {
+    findRef: source => {
+      const refs = api.getRefs();
+
+      return Object.values(refs).find(({ url }) => `${url}/iframe.html`.match(source));
+    },
+    changeRefVersion: (id, url) => {
+      const previous = api.getRefs()[id];
+      const ref = { ...previous, stories: {}, url } as SetRefData;
+
+      api.checkRef(ref);
+    },
+    checkRef: async ref => {
+      const { id, url } = ref;
+
+      const response = await fetch(`${url}/data.json`).catch(() => false);
+
+      if (response) {
+        const { ok } = response;
+
+        if (ok) {
+          const data: SetRefData = await response
+            .json()
+            .catch((error: Error) => ({ startInjected: true, error }));
+
+          api.setRef(id, { id, url, startInjected: false, ...data });
+        } else {
+          api.setRef(id, { id, url, startInjected: true });
+        }
+      } else {
+        logger.warn('an auto-injected ref threw a cors-error');
+        api.setRef(id, { id, url, startInjected: true });
+      }
+    },
+
+    getRefs: () => {
+      const { refs: fromConfig = {} } = provider.getConfig();
+      const { refs: fromState = {} } = store.getState();
+
+      return { ...fromConfig, ...fromState };
+    },
+
+    setRef: (id, { stories, ...rest }, ready = false) => {
+      const ref = api.getRefs()[id];
+      const after = stories
+        ? namespace(
+            transformStoriesRawToStoriesHash(
+              map(stories, ref, { mapper: defaultMapper }),
+              {},
+              { provider }
+            ),
+            ref
+          )
+        : undefined;
+
+      const result = { ...ref, stories: after, ...rest, ready };
+
+      store.setState({
+        refs: {
+          ...(store.getState().refs || {}),
+          [id]: result,
+        },
+      });
+    },
+  };
+
+  const refs = Object.entries(api.getRefs());
```

Comment:
```suggestion
  const refs = Object.entries(provider.getConfig().refs);
```

This is the point where we move the static refs into state.